build-and-test.ps1

```diff
@@ -1,36 +1,56 @@
 #!/usr/bin/env pwsh
-[cmdletbinding()]
+[cmdletbinding(
+    DefaultParameterSetName = "BuildAndTest"
+)]
 param(
-    [ValidateSet('runtime', 'sdk', 'aspnet', 'wcf')]
+    [ValidateSet("runtime", "sdk", "aspnet", "wcf")]
     [string[]]$RepoFilter = @(),
     [string]$VersionFilter = "*",
     [string]$OSFilter = "*",
-    [string]$OptionalImageBuilderArgs,
-    [switch]$SkipTesting = $false
+    [Parameter(ParameterSetName = "Build")]
+    [switch]$BuildOnly,
+    [Parameter(ParameterSetName = "Test")]
+    [switch]$TestOnly,
+    [Parameter(ParameterSetName = "Build")]
+    [Parameter(ParameterSetName = "BuildAndTest")]
+    [string]$OptionalImageBuilderArgs
 )
 
+if ($PSCmdlet.ParameterSetName -eq "BuildAndTest") {
+    $build = $true
+    $test = $true
+}
+else {
+    $build = $BuildOnly
+    $test = $TestOnly
+}
+
 if ($RepoFilter.Count -eq 0) {
     $PathFilters = $null
-    $optionalTestArgs = ""
+    $testCategories = @("runtime", "sdk", "aspnet", "wcf")
 }
 else {
     $PathFilters = ""
     $RepoFilter | foreach {
         $PathFilters += " --path '$VersionFilter/$_/$OSFilter'"
     }
-    # Convert the array to a comma-delimited string of the repos with each repo value in quotes
-    # (e.g. "runtime", "sdk")
-    $repoList = ($RepoFilter | foreach { "`"$_`"" }) -join ", "
-    $optionalTestArgs = "-TestCategories @($repoList)"
+    $testCategories = $RepoFilter
 }
 
 $OptionalImageBuilderArgs += " --manifest manifest.json"
```

Comment:
Thoughts on removing this?  Is it necessary?  Doesn't it default to this?  It isn't specified in the core repo.

Reply:
Makes sense.  I think this was leftover from when we had separate manifest for aspnet/wcf.